presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcPageSourceFactory.java

```diff
@@ -229,7 +229,7 @@ private static String splitError(Throwable t, Path path, long start, long length
 
     private static void verifyFileHasColumnNames(List<String> physicalColumnNames, Path path)
     {
-        if (physicalColumnNames.stream().allMatch(physicalColumnName -> DEFAULT_HIVE_COLUMN_NAME_PATTERN.matcher(physicalColumnName).matches())) {
+        if (physicalColumnNames.size() > 0 && physicalColumnNames.stream().allMatch(physicalColumnName -> DEFAULT_HIVE_COLUMN_NAME_PATTERN.matcher(physicalColumnName).matches())) {
```

Comment:
This can be `!physicalColumnNames.isEmpty()` which for some collections is faster